api/client.go

```diff
@@ -399,6 +405,9 @@ func (c *Client) NewRequest(method, requestPath string) *Request {
 	if c.config.Timeout != 0 {
 		c.config.HttpClient.Timeout = c.config.Timeout
 	}
+	if c.headers != nil {
```

Comment:
I think that this actually needs to iterate through and *add* headers, not replace the whole header structure unilaterally.

Reply:
I think this works now because the ToHTTP method that is called from RawRequest already does that if any headers in the passed in request is set: [request.go](https://github.com/hashicorp/vault/blob/44e0deff4490b32fa6dd4cd9392a4f7123142e48/api/request.go#L64). Before these changes no headers were set before ToHTTP was called.

It might be better to do as you suggested though. Let me know what you think and I'll update accordingly!

Reply:
Ah, no -- that looks fine. Thanks for clarifying!

Reply:
Great! No problem, thanks for merging this!